pkg/store/proxy.go

```diff
@@ -302,14 +305,17 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
 		return nil
 	}
 
+	var unsortedSeriesSetDetected bool
 	storeResponses := make([]respSet, 0, len(stores))
-
 	for _, st := range stores {
 		st := st
+		if !st.SendsSortedSeries() {
```

Comment:
This was added here: https://github.com/thanos-io/thanos/pull/5709

In theory we can also try to detect whether a store sends sorted series, but that might just add a bit more complexity.

Reply:
@bwplotka @GiedriusS We can also add a check for sorted series in the [seriesServer](https://github.com/thanos-io/thanos/blob/main/pkg/query/querier.go#L164) and avoid adding detections throughout the codebase. Since we are fetching series one by one, we can always keep the previous one and do a label comparison without replica labels.

Reply:
I don't think we should add it to `seriesServer` because I think it will have to be removed in general for `Select()` to be fully streamed because `seriesServer` is acting as a buffer right now and IMHO we should return the streamed heap directly once `sortDedupLabelset` is gone.

Reply:
Ok, sounds good to me. Once stores start to send series with replica labels at the end, we can remove all checks and detections. We could also document it as an explicit requirement of the store API so that we don't have to worry about external implementations.